### ISSUE TITLE
Bug: Unfrozen Bullet Bills Remain Stationary

### ISSUE THREAD

**Issue by ChristianSilvermoon** [Wohlstand/TheXTech#72]
If you freeze a Bullet Bill using an Ice Flower, and then Unfreeze it using a Fire Flower, it remains stationary.
![Scr_2021-05-16_12-23-29](https://user-images.githubusercontent.com/30162808/118410291-79f38a80-b643-11eb-8475-634d3c125bc9.gif)

GIF is from [ds-sloth's repo @ 92d12fd](https://github.com/ds-sloth/TheXTech/commit/92d12fd4abbfa9e8edbb7bd64132275f0c382c45)

but issue appears to originate from here

**Comment by Wohlstand** [comment]
Yeah, I can confirm:
![Scr_2021-02-10_01-32-50](https://user-images.githubusercontent.com/6751442/118413803-cd98bf00-b6a9-11eb-950e-4eee831a5427.gif)

It's my bug, at the vanilla game it works fine:
![Peek 2021-05-16 22-33](https://user-images.githubusercontent.com/6751442/118413820-f0c36e80-b6a9-11eb-8ab7-b6ab17b81947.gif)

**Comment by PatriotRossii** [comment]
I wanna try to dive into this and fix it

**Comment by ds-sloth** [comment]
The truth is even stranger than expected.
The direction of the bullet was being passed by reference to `EditorNPCFrame` when it was unfrozen, and `EditorNPCFrame` randomly sets the direction when it is zero.
I have updated `EditorNPCFrame` to take its direction parameter by reference.
I assume that was not what you expected, so I have also added a compat setting (on by default) that makes the bullet bill cannon set the bullet bill's `DefaultDirection` to its `Direction`, making it always keep its direction when it is melted.

Vanilla behavior:
![Scr_2021-06-02_01-27-14](https://user-images.githubusercontent.com/72112344/120428824-5a967600-c342-11eb-81e1-e32b6f764322.gif)

New default behavior:
![Scr_2021-06-02_01-28-12](https://user-images.githubusercontent.com/72112344/120428846-62eeb100-c342-11eb-8912-bcc1e28f507b.gif)
